src/api/server.ts

```diff
@@ -0,0 +1,156 @@
+import express from 'express';
+import http from 'http';
+import { WebSocketServer, WebSocket } from 'ws';
+import { WebSocketEventSubscriber } from './websocket-subscriber.js';
+import { MCPClientManager } from '../client/manager.js';
+import { ILLMService } from '../ai/llm/services/types.js';
+import { logger } from '../utils/logger.js';
+import { EventEmitter } from 'events';
+import { StreamableHTTPServerTransport } from '@modelcontextprotocol/sdk/server/streamableHttp.js';
+import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
+import { z } from 'zod';
+import { randomUUID } from 'crypto';
+
+export async function initializeApi(
+    clientManager: MCPClientManager,
+    llmService: ILLMService,
+    agentEventBus: EventEmitter
+) {
+    const app = express();
+    const server = http.createServer(app);
+    const wss = new WebSocketServer({ server });
+
+    // set up event broadcasting over WebSocket
+    const webSubscriber = new WebSocketEventSubscriber(wss);
+    logger.info('Setting up API event subscriptions...');
+    webSubscriber.subscribe(agentEventBus);
+
+    // HTTP endpoints
+    app.post('/api/message', express.json(), async (req, res) => {
```

Comment:
we should rethink API names, but we can defer this for later. let's add a TODO comment at the top of this function that API names are work in progress and might be refactored/renamed